### DISCUSSION TITLE
Better way to handle massive files?

### DISCUSSION THREAD

**Discussion by Lexian-droid** [rahulhaque/laravel-filepond#56] [Q&A] [answered]
Hello, I am working on a project that uses large files however it just occured to me that anyone anywhere can upload these very large files as well as the legitimate users,

Currently, my global filepond config is set to:
```php
    'validation_rules' => [
        'required',
        'file',
        'max:8589934592',
    ]
```

I was wondering if there was any better way to handle this, for example maybe specify a specific endpoint that I can upload very large files to instead of the default route?

all help appreciated, thanks in advance!

**Comment by rahulhaque** [comment]
@Lexian-droid first use filepond's frontend file size validation. Also, did you try chunk upload for large file upload? If you do, then set the max file size in config to max chunk size you wish to allow for large file uploads. Non-chunked upload will respect the max file size as well. There is also controller level validation available before you allow a user to move the file to permanent storage, see [here](https://github.com/rahulhaque/laravel-filepond/blob/10.x/README.md#rulefilepondrules).

**Reply by Lexian-droid** [reply]
Hello, thank you so much for your response.

I see, the problem is, I am using default controllers and scripts to accomplish the uploading, so I don't know about if it's "Chunk uploading" or not, do you by chance know if this is true or not?

If it's not using chunk uploading, do you care to point me to where I can configure it to do so? as I am not 100% sure where to start.
thanks!

**Reply by rahulhaque** [reply, marked as answer]
See filepond's `chunkUploads` implementation [here](https://pqina.nl/filepond/docs/api/instance/properties/#server).

Edit: What I meant by the above is that you should use `chunkUploads` true when initiating filepond.

**Reply by Lexian-droid** [reply]
I see, alright, I have that working now, thank you!

I know this is unrelated, I'm okay with opening a new discussion, but I have another issue, I've got another field on the page that is meant for assets, so like images, videos, plain text, etc, I am having issues with uploading files from this one, because I am getting the error message 500 ``ValueError: Path cannot be empty in file ...``

Do you by chance know what's going on there? or shall we move it to a different discussion?

Edit: Here is my current script:
```html
                        <form action="{{ route('applications.assets.create', ['appid' => $application->id]) }}" method="POST" enctype="multipart/form-data">
                            @csrf
                            <div class="form-group">
                                <label for="file">File:</label>
                                <input type="file" name="file" id="file" class="w3-input form-control-file" accept="image/*, video/*, text/plain">
                            </div>
                            <button type="submit" class="w3-margin-top w3-margin-bottom w3-button w3-border w3-hover-blue">Submit</button>
                        </form>
                        <script>
                            // Set default FilePond options
                            FilePond.setOptions({
                                server: {
                                    url: "{{ config('filepond.server.url') }}",
                                    headers: {
                                        'X-CSRF-TOKEN': "{{ @csrf_token() }}",
                                    },
                                },
                            });

                            // Create a FilePond instance
                            FilePond.create(document.querySelector('input[type="file"]'));
                        </script>
```

**Reply by rahulhaque** [reply]
I don't see anything wrong in this code. Need more context and backend code of file retrieval to understand what causing the error.

**Reply by Lexian-droid** [reply]
> I don't see anything wrong in this code. Need more context and backend code of file retrieval to understand what causing the error.

Hello, what's strange is that the code works on my production server, but it wasn't working on my local server, I figured out the issue, I had to put a tmp file path in php.ini, I am on Windows so maybe PHP didn't have the ability to write to the default path? but regardless, it's fixed now :) I appreciate the help, apologies if it was a waste of time.

**Reply by rahulhaque** [reply]
Glad that you solved it. Thank you for using my package.
